cranelift/codegen/src/ir/pcc.rs

```diff
@@ -863,6 +1269,53 @@ impl<'a> FactContext<'a> {
         }
         Ok(())
     }
+
+    /// Apply a known inequality to rewrite dynamic bounds using transitivity, if possible.
+    ///
+    /// Given that `lhs >= rhs` (if not `strict`) or `lhs < rhs` (if
+    /// `strict`), update `fact.
+    pub fn apply_inequality(&self, fact: &Fact, lhs: &Fact, rhs: &Fact, strict: bool) -> Fact {
+        trace!(
+            "lhs single expr = {:?} rhs single expr = {:?}",
+            lhs.is_single_expr(),
+            rhs.is_single_expr()
+        );
+        match (lhs.is_single_expr(), rhs.is_single_expr(), fact) {
+            (
+                Some(lhs),
+                Some(rhs),
+                Fact::DynamicMem {
+                    ty,
+                    min,
+                    max,
+                    nullable,
+                },
+            ) if rhs.base == max.base => {
+                trace!("bases match");
```

Comment:
Seems like maybe not the most useful trace long term.

Reply:
Removed; I neglected to go through and clean these up after getting things working, sorry!